pkg/agent/datamodel/types.go

```diff
@@ -681,6 +682,11 @@ type SysctlConfig struct {
 	VMVfsCachePressure             *int32 `json:"vmVfsCachePressure,omitempty"`
 }
 
+type UlimitConfig struct {
+	MaxLockedMemory string `json:"memlock,omitempty"`
```

Comment:
your json tags match the OS settings but I don't see you actually using this for serialization anywhere

I would change these because we do serve an HTTP API so on the wire between RP <-> AB this is how RP will serialize it.

you can have a separate type for on disk serialization if you later decide to use that approach